code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -80,16 +80,16 @@
 	if (target == loc || !target || action_busy)
 		return
 
-	if(!do_after(src, 12, TRUE, 5, BUSY_ICON_HOSTILE))
-		return
-
 	if (used_acid_spray || !check_plasma(200))
 		return
 
 	if(stagger)
 		to_chat(src, "<span class='xenowarning'>The shock disrupts you!</span>")
 		return
 
+	if(!do_after(src, 5, TRUE, 5, BUSY_ICON_HOSTILE))
+		return
```

Comment:
We want stagger to be able to disrupt the action. That is why the check should be after this.

Reply:
I didn't see the earlier stagger check before it; fixed.